### PR TITLE
[local-up-cluster] Update docker/containerd/runc versions and capture more logs

### PR DESCRIPTION
**Pull request by dims**
#### What type of PR is this?
/kind cleanup

- use apt to install fresh stuff
- print versions of the components
- throw on debug for docker log
- capture dmesg log
- bump kernel logging

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
thanks @ameukam 

/hold cancel

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -695,6 +695,31 @@ function wait_node_ready(){
   fi
 }
 
+function refresh_docker_containerd_runc {
+  apt update
```
Comment (BenTheElder) [inline hack/local-up-cluster.sh:699]: requiring debian-ish is a pretty big departure isn't it?

Comment (BenTheElder) [reply hack/local-up-cluster.sh:699]: perhaps we should update these packages in the host environment instead? (IE kubekins-e2e)

Comment (dims) [reply hack/local-up-cluster.sh:699]: @BenTheElder this piece of code will not be executed outside of the CI

**Pull request merged**